packages/labs/src/components/select/suggest.tsx

```diff
@@ -0,0 +1,265 @@
+                popoverClassName={classNames(Classes.SELECT_POPOVER, popoverProps.popoverClassName)}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillClose={this.handlePopoverWillClose}
+            >
+                <div
+                    onKeyDown={this.getTargetKeyDownHandler(handleKeyDown)}
+                    onKeyUp={this.state.isOpen ? handleKeyUp : undefined}
+                    onClick={this.handleInputClick}
+                >
+                    <InputGroup
+                        placeholder="Search..."
+                        value={inputValue}
+                        {...htmlInputProps}
+                        inputRef={this.refHandlers.input}
+                        onChange={this.handleQueryChange}
+                    />
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <Menu ulRef={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </Menu>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
+        }));
+    }
+
+    private selectText = () => {
+        if (this.input != null) {
+            this.input.focus();
+            // wait until the input is focused to select the text inside of it
+            requestAnimationFrame(() => this.input.setSelectionRange(0, this.input.value.length));
+        }
+    }
+
+    private handleInputClick = (_event: React.SyntheticEvent<HTMLElement>) => {
+        this.selectText();
+    }
+
+    private handleActiveItemChange = (activeItem: T) => this.setState({ activeItem });
+
+    private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
+        this.selectText();
+
+        this.setState({
+            isTyping: false,
+            query: "",
+            selectedItem: item,
+        });
+
+        Utils.safeInvoke(this.props.onItemSelect, item, event);
+    }
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => requestAnimationFrame(() => {
+        const { closeOnSelect, popoverProps = {} } = this.props;
+
+        if (this.input == null) {
+            return;
+        }
+
+        if (this.input !== document.activeElement) {
+            // the input is no longer focused so we can close the popover
+            this.input.blur();
+            this.setState({ isOpen: false });
+        } else {
+            this.setState({ isOpen: closeOnSelect ? nextOpenState : true });
```

Comment:
I'm not sure what you mean by no-op, but your suggestion is not equivalent to what I have. 

This logic is messed up anyway, I'm gonna handle `closeOnSelect` stuff where it belongs: `handleItemSelect`